src/ooo_cpu.cc

```diff
@@ -346,37 +393,29 @@ void O3_CPU::fetch_instruction()
 	  if(rq_index != -2)
 	    {
 	      // successfully sent to the ITLB, so mark all instructions in the IFETCH_BUFFER that match this ip as translated INFLIGHT
+	      uint32_t translate_inflight_index = index;
 	      for(uint32_t j=0; j<IFETCH_BUFFER.SIZE; j++)
 		{
-                    if(((IFETCH_BUFFER.entry[j].ip >> LOG2_PAGE_SIZE) == (ifb_entry.ip >> LOG2_PAGE_SIZE)) && (IFETCH_BUFFER.entry[j].translated == 0))
-		    {
-		      IFETCH_BUFFER.entry[j].translated = INFLIGHT;
-		      IFETCH_BUFFER.entry[j].fetched = 0;
-		    }
+		    if(IFETCH_BUFFER.entry[translate_inflight_index].ip == 0)
+		      {
+			break;
+		      }
```

Comment:
This check can safely be removed, since the if-statement on the next line will filter it out.

Reply:
I added it as a performance optimization to actually get out of the for-loop, instead of scanning the entire IFETCH_BUFFER every time.  This loop and ones like it take up more and more time for larger IFETCH_BUFFER sizes.  I put in similar code in other for-loops that scan the IFETCH_BUFFER.